### PR TITLE
Deliver Iter 3 Features

### PR DESCRIPTION
**Pull request by AlanHe-Xiaoyu**
This PR includes the three features below:

1. Feature remove-order from order view  
2. Feature apply effect to individual item in orders view  
3. Feature de-duplicate order-level comments from individual item comments (this might need a refresh when using a browser because of cached info) 
You can try out the features on:

https://rocky-ravine-27276.herokuapp.com/customers/1

### CODE REVIEW

**Comment by Hexhu** [comment]
> Since you have made the check by adding line 22, do we still to check the content below?

The check on line 22 querys the Items table directly rather than inspecting the contents of the webpage. I think they are two different tests.

File: features/orders/view_orders.feature [hunk 1]
```diff
@@ -20,8 +20,8 @@ Scenario: view correct order info
   Then I should see the following details for that order:
   | content                           |
   | Purchaser: Tom Foolery            |
-  | Comments: Pickup by: Al Foolery   |
   | Gift? No                          |
   | Order total: $19.00 (Credit card) |
-
+  And I should not see the following details for that order:
```
Comment (armandofox) [inline features/orders/view_orders.feature]: Be careful with negative expectations. If the page shows a random comment or a different comment or even an error message, this step will still pass.  In this case it's less of an issue so you don't need to chagne it, but in general if you are checking for the *absence* of a field, you can also use `expect(page).not_to have_selector(some_css_expression)` or something similar.

File: features/step_definitions/order_steps.rb [hunk 2]
```diff
@@ -48,6 +48,14 @@ def buy!(customer, vtype, qty, showdate=nil)
   end
 end
 
+Then /^I should not see the following details for that order:$/ do |table|
+  within("div#details_order_#{@order.id}") do
+    table.hashes.each do |h|
+      page.should_not have_content h[:content]
```
Comment (armandofox) [inline features/step_definitions/order_steps.rb]: we're trying to migrate to the RSpec 3 syntax of `expect(page).not_to have_content()` - could you change to that please? thx.

File: features/reservations/staff_comment_on_reservation.feature [hunk 3]
```diff
@@ -9,18 +9,6 @@ Background: customer with existing reservation
 
   Given I am logged in as boxoffice manager
 
-Scenario: add comment to revenue reservation
```
Comment (armandofox) [inline features/reservations/staff_comment_on_reservation.feature]: Why was this scenario deleted?

Comment (autumnli11) [reply features/reservations/staff_comment_on_reservation.feature]: I think we deleted this scenario to reflect the fact that order level comment no longer exists (if we understood this test correctly).

Comment (autumnli11) [reply features/reservations/staff_comment_on_reservation.feature]: and the new scenario we added checks item level comment.

File: features/step_definitions/order_steps.rb (hunk 2 above)
Comment (AlanHe-Xiaoyu) [reply features/step_definitions/order_steps.rb]: Okay will do so now.

File: features/orders/view_orders.feature (hunk 1 above)
Comment (AlanHe-Xiaoyu) [reply features/orders/view_orders.feature]: Got it. I'll change it now.

**Pull request merged**